Lib/logging/__init__.py

```diff
@@ -883,25 +875,20 @@ def _removeHandlerRef(wr):
     # set to None. It can also be called from another thread. So we need to
     # pre-emptively grab the necessary globals and check if they're None,
     # to prevent race conditions and failures during interpreter shutdown.
-    acquire, release, handlers = _acquireLock, _releaseLock, _handlerList
-    if acquire and release and handlers:
-        acquire()
-        try:
-            handlers.remove(wr)
-        except ValueError:
-            pass
-        finally:
-            release()
+    handlers = _handlerList
```

Comment:
In light of this comment (which I missed earlier), should we also capture the ``lock`` variable to the function ``locals()``? (@vstinner might have insights here w.r.t. module teardown)

A

Reply:
Have updated to assign _lock to local variable lock, as per @vstinner 's suggestion